src/dom/ui-manip.js

```diff
@@ -0,0 +1,106 @@
+import { format, parse } from "date-fns";
+
+const spinnerContainer = document.getElementById("all");
+const detailsContainer = document.getElementById("heading");
+const buttonContainer = document.getElementById("select-button");
+const racesContainer = document.getElementById("select-race");
+const lapDetailsContainer = document.getElementById("lap-details");
+
+export default function addLapButton(btnNum, runsJSON) {
+  const button = document.createElement("button");
+
+  button.id = `${btnNum}`;
+  button.innerHTML = `
+  <div>
+    Lap ${btnNum}: Time: 
+    ${runsJSON.lapSummaries[btnNum - 1]["time lap"] / 1000}s - 
+    Max Speed: ${runsJSON.lapSummaries[btnNum - 1]["Max Speed GPS"] / 10}km/h
+  </div>
+  `;
+
+  buttonContainer.appendChild(button);
+
+  return button;
+}
+
+export function showSpinner() {
+  spinnerContainer.style = `opacity: 0.7;`;
+}
+
+export function hideSpinner() {
+  spinnerContainer.style = `opacity: 1.0;`;
+}
+
+export function addHeaderDetails(runsJSON, lapNum) {
+  detailsContainer.id = "details";
+  detailsContainer.innerHTML = `${runsJSON.trackName}:<br />${runsJSON.driver} - Lap ${lapNum}`;
+}
+
+export function addLapDetails(runsJSON, lapNum) {
+  lapDetailsContainer.innerHTML = `
+
+    <h3>Lap:  <span>${lapNum}</span></h3>
```

Comment:
Note for tailwind
h3's all have the same semantic hierarchy - meaning they are all subtopics of an h2 which is in it of itself a subtopic of the page, titled by an h1

My suggestion is to use <span> with tailwind classes or just divs